### PR TITLE
Build: scripts to build and measure packages reference docs

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
Scripts that can be used to start builds and check the code coverage metrics of the packages docs.

https://github.com/grafana/grafana/issues/22957

### CODE REVIEW

File: scripts/ci-reference-docs-metrics.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+WARNINGS_COUNT_LIMIT=900
+
+if [ "$WARNINGS_COUNT" -gt $WARNINGS_COUNT_LIMIT ]; then
+  echo -e "API Extractor warnings/errors $WARNINGS_COUNT exceeded $WARNINGS_COUNT_LIMIT so failing build.\n"
+  echo -e "Please go to: https://github.com/grafana/grafana/blob/master/contribute/style-guides/code-comments.md for more information on how to add code comments."
+  exit 1
+fi
+
+if [ "$WARNINGS_COUNT" -lt $WARNINGS_COUNT_LIMIT ]; then
+  echo -e "Wohoo! Less number of warnings compared to last build 🎉🎈🍾✨\n\nYou can lower the threshold from $WARNINGS_COUNT_LIMIT to $WARNINGS_COUNT in the:\nscripts/ci-reference-docs-metrics.sh"
```
Comment (dprokop) [inline scripts/ci-reference-docs-metrics.sh:18]: 👍

**Review by dprokop — CHANGES_REQUESTED** [review]
This is great! 💯 Can you also push the metrics to Graphite? This should happen on master only. Think you can add that to ci-frontend-metrics.sh script.

File: scripts/ci-reference-docs-metrics.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# abort if we get any error
+set -e
```
Comment (aknuds1) [inline scripts/ci-reference-docs-metrics.sh:4]: Adding `o pipefail` should ensure pipelines also fail I think if one of the steps fails.
```suggestion
set -eo pipefail
```

File: scripts/ci-reference-docs-metrics.sh (hunk 1 above)
Comment (aknuds1) [inline scripts/ci-reference-docs-metrics.sh:18]:
```suggestion
  echo -e "Wohoo! Fewer warnings compared to last build 🎉🎈🍾✨\n\nYou can lower the threshold from $WARNINGS_COUNT_LIMIT to $WARNINGS_COUNT in the:\nscripts/ci-reference-docs-metrics.sh"
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]

File: scripts/ci-reference-docs-metrics.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  echo -e "Please go to: https://github.com/grafana/grafana/blob/master/contribute/style-guides/code-comments.md for more information on how to add code comments."
+  report_reference_docs_metrics "$CIRCLE_BRANCH" "$WARNINGS_COUNT"
+  exit 1
+fi
+
+if [ "$WARNINGS_COUNT" -lt $WARNINGS_COUNT_LIMIT ]; then
+  pretty_print_result_of_report "Wohoo! Fewer warnings compared to last build 🎉🎈🍾✨\n\nYou can lower the threshold from $WARNINGS_COUNT_LIMIT to $WARNINGS_COUNT in the:\nscripts/ci-reference-docs-metrics.sh"
+  report_reference_docs_metrics "$CIRCLE_BRANCH" "$WARNINGS_COUNT"
+  exit 0
+fi
```
Comment (hugohaggmark) [inline scripts/ci-reference-docs-metrics.sh:49]: Nice touch!

**Review by hugohaggmark — APPROVED** [review]
Brilliant work on this! 🌟

File: scripts/ci-reference-docs-metrics.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# abort if we get any error
+set -eo pipefail
+
```
Comment (aknuds1) [inline scripts/ci-reference-docs-metrics.sh:5]: Don't need this extra blank line.
```suggestion
```

File: scripts/ci-reference-docs-metrics.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  echo -e "-----------------------------------------------------\n"
+  echo -e "$1\n"
+  echo -e "-----------------------------------------------------"
+}
+
+REPORT_PATH="$(realpath "$(dirname "$0")/../reports/docs/")"
+BUILD_SCRIPT_PATH="$(realpath "$(dirname "$0")/ci-reference-docs-build.sh")"
+
+if [ ! -d "$REPORT_PATH" ]; then
+  # this script needs to be run after the packages have been build and the api-extractor has completed.
```
Comment (aknuds1) [inline scripts/ci-reference-docs-metrics.sh:30]:
```suggestion
  # this script needs to be run after the packages have been built and the api-extractor has completed.
```

**Review by aknuds1 — APPROVED** [review]
LGTM - just fixed a typo.

**Comment by aknuds1** [comment]
Looks as if CI fails consistently: https://app.circleci.com/pipelines/github/grafana/grafana/10669/workflows/d946c5af-f6cc-4459-abe6-8bbe975e8735/jobs/467490.

**Comment by mckn** [comment]
> Looks as if CI fails consistently: https://app.circleci.com/pipelines/github/grafana/grafana/10669/workflows/d946c5af-f6cc-4459-abe6-8bbe975e8735/jobs/467490.

ah, yeah! I will fix that one :)

**Review by dprokop — APPROVED** [review]
🎉

**Pull request merged**